src/hooks/init/check-update.ts

```diff
@@ -1,69 +1,135 @@
 import {Hook} from '@oclif/core'
-import type {Chalk} from 'chalk'
-import {spawn} from 'child_process'
-import {readFile, stat} from 'fs/promises'
-import {join} from 'path'
+import chalk from 'chalk'
+import makeDebug from 'debug'
+import {spawn} from 'node:child_process'
+import {Stats} from 'node:fs'
+import {readFile, stat, utimes} from 'node:fs/promises'
+import {dirname, join, resolve} from 'node:path'
+import {fileURLToPath} from 'node:url'
+
+const readJSON = async <T>(file: string): Promise<{contents: T; stat: Stats}> => ({
+  contents: JSON.parse(await readFile(file, 'utf8')),
+  stat: await stat(file),
+})
```

Comment:
perf: promise.all the 2 tasks, then return them in the object.

[small perf nit, but this runs before every command]